### ISSUE TITLE
Allow redirect instead of iframe for preview

### ISSUE THREAD

**Issue by jaap3** [torchbox/wagtail-headless-preview#15]
Our sites use X-Frame-Options to deny being iframed (for all the good reasons). However, this breaks the preview feature of wagtail-headless-preview.

I'm not entirely sure why an iframe is used, but I assume there are reasons. Would a PR where a redirect is used in serve_preview (based om some setting) be appreciated/considered?

**Comment by zerolab** [comment]
In a non-headless setup the preview path uses the relevant template with the fake page data. As we do not have control over the front-end, using an iframe took least assumptions.

Happy for setting-based PR. You will want to test how live preview plays with it.

**Comment by jaap3** [comment]
Live preview will probably not work if a redirect is used. I couldn't really come up with a good setting/docs so I've decided to not provide a patch and will instead explicitly allow the preview page to be iframed in our frontend application.

**Comment by zerolab** [comment]
@jaap3 we oculd still support redirects, with the caveat that it won't support live preview. A few other people asked for redirect in the Wagtail Slack so it is not an uncommon need.

**Comment by jaap3** [comment]
I've reopened this. What I came up with was basically adding:

```
if getattr(settings, 'HEADLESS_PREVIEW_REDIRECT', False):
    response = redirect(self.get_preview_url(response_token))  # from django.shortcuts import redirect
else:
```

before https://github.com/torchbox/wagtail-headless-preview/blob/e16d10755e6846f03caf5ed805060444831745f9/wagtail_headless_preview/models.py#L114-L118

But then I got in my head about conflicting settings and what to do about it (i.e. enabling both `HEADLESS_PREVIEW_REDIRECT` and `HEADLESS_PREVIEW_LIVE`) and decided that it was probably easier to fix things on my side.
